src/pages/Group/GroupFeed.tsx

```diff
@@ -0,0 +1,184 @@
+import { Group } from "../../data/models/Group";
+import Spinner from "../../components/Spinner";
+import { useApi } from "../../contexts/ApiProvider";
+import { useInfiniteQuery, useQuery } from "react-query";
+import { QueryKeys } from "../../constants";
+import { Happiness, HappinessPagination } from "../../data/models/Happiness";
+import FeedCard from "./FeedCard";
+import HappinessViewerModal from "../../components/modals/HappinessViewerModal";
+import React, { useEffect, useMemo, useState } from "react";
+import { useInView } from "react-intersection-observer";
+import { dateFromStr, formatDate, modifyDateDay } from "../../utils";
+
+// Returns "today" or "yesterday" if the given date is today or yesterday
+// Otherwise returns the default string
+export const dateOrTodayYesterday = (date: string, otherwise: string) => {
+  if (date === formatDate(new Date())) return "Today";
+  else if (date === formatDate(modifyDateDay(new Date(), -1)))
+    return "Yesterday";
+  else return otherwise;
+};
+
+// The group feed first shows the user's unread entries in the group (which are marked as read as they are scrolled past)
+// followed by an infinite scroll of all the group's entries
+export default function GroupFeed({ groupData }: { groupData: Group }) {
+  const { api } = useApi();
+  const [selectedEntry, setSelectedEntry] = useState<Happiness>();
+
+  // for infinite scroll
+  const [bottomRef, bottomInView] = useInView();
+
+  // query for fetching unread entries
+  const unreadQuery = useQuery<Happiness[]>({
+    queryKey: [QueryKeys.FETCH_GROUP_HAPPINESS_UNREAD, groupData.id],
+    queryFn: () =>
+      api
+        .get<Happiness[]>(`/group/${groupData.id}/happiness/unread`)
+        .then((res) => res.data),
+    cacheTime: 0, // disable cache since this causes stale reads to be shown on initial load
```

Comment:
I suppose this is a fine solution, but in the future I think we should look into using `setQueryData` to update data when an entry is read.

Reply:
no I tried that but it doesn't work: setQueryData immediately updates the data, so the UI would change. But remember we don't want that to happen since we alr have the New tag and reads are just internal state.